### PR TITLE
Manage cdo fixes

### PR DESCRIPTION
**Pull request by jbarnsley10**
Cookie prefs now works
Added extra tests
Fixed back nav on Manage CDO

### CODE REVIEW

File: app/api/ddi-index-api/cdo.js [hunk 1]
```diff
@@ -34,12 +34,8 @@ const getCdoTaskDetails = async (indexNumber, taskName) => {
...
 }
 
-const taskNameMapper = {
-  'send-application-pack': 'sendApplicationPack'
-}
-
-const saveCdoTaskDetails = async (indexNumber, taskName, payload, user) => {
-  const res = await post(`${cdoEndpoint}/${indexNumber}/manage:${taskNameMapper[taskName]}`, payload, user)
+const saveCdoTaskDetails = async (indexNumber, apiKey, payload, user) => {
+  const res = await post(`${cdoEndpoint}/${indexNumber}/manage:${apiKey}`, payload, user)
```
Comment (whitewaterdesign) [inline app/api/ddi-index-api/cdo.js:38]: Thanks.  This much nicer. 👍

File: app/routes/cdo/manage/tasks/generic-task.js [hunk 2]
```diff
@@ -55,7 +60,10 @@ module.exports = [{
       const dogIndex = request.params.dogIndex
       const taskName = request.params.taskName
 
-      await saveCdoTaskDetails(dogIndex, taskName, request.payload, getUser(request))
+      const { apiKey } = getTaskDetailsByKey(taskName)
+      console.log('taskName', taskName)
+      console.log('apiKey', apiKey)
+      await saveCdoTaskDetails(dogIndex, apiKey, request.payload, getUser(request))
```
Comment (whitewaterdesign) [inline app/routes/cdo/manage/tasks/generic-task.js:66]: console logs will need removing

Comment (jbarnsley10) [reply app/routes/cdo/manage/tasks/generic-task.js:66]: Doh! Of course. Forgot about those

File: app/models/cdo/manage/tasks/record-application-fee-payment.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const { constructDateField } = require('../../../../lib/model-helpers')
 
 function ViewModel (data, backNav, errors) {
   this.model = {
-    backLink: backNav.backLink,
+    backLink: backNav.backLink === '/' ? `/cdo/manage/cdo/${data.indexNumber}` : backNav.backLink,
     srcHashParam: backNav.srcHashParam,
```
Comment (whitewaterdesign) [inline app/models/cdo/manage/tasks/record-application-fee-payment.js:7]: 👍

File: app/routes/cdo/manage/tasks/generic-task-helper.js [hunk 4]
```diff
@@ -15,12 +15,12 @@ const { getCompanies } = require('../../../../api/ddi-index-api/insurance')
...
-  { name: tasks.form2Sent, Model: ViewModelSendForm2, validation: validateSendForm2, key: 'send-form2', label: 'Send Form 2' },
-  { name: tasks.verificationDateRecorded, Model: ViewModelRecordVerificationDates, validation: validateVerificationDates, key: 'record-verification-dates', label: 'Record the verification date for microchip and neutering' }
+  { name: tasks.applicationPackSent, Model: ViewModelSendApplicationPack, validation: validateSendApplicationPack, key: 'send-application-pack', label: 'Send application pack', apiKey: 'sendApplicationPack', stateKey: 'applicationPackSent' },
+  { name: tasks.insuranceDetailsRecorded, Model: ViewModelRecordInsuranceDetails, validation: validatePayloadRecordInsuranceDetails, key: 'record-insurance-details', label: 'Record insurance details', apiKey: 'recordInsuranceDetails', stateKey: 'insuranceDetailsRecorded' },
+  { name: tasks.microchipNumberRecorded, Model: ViewModelRecordMicrochipNumber, validation: validateMicrochipNumber, key: 'record-microchip-number', label: 'Record microchip number', apiKey: 'microchipNumber', stateKey: 'microchipNumberRecorded' },
```
Comment (whitewaterdesign) [inline app/routes/cdo/manage/tasks/generic-task-helper.js]: can you change this one to ``recordMicrochipNumber``?   The apiKey we agreed wasn't consistent with other endpoints.

File: app/routes/cdo/manage/tasks/generic-task-helper.js [hunk 5]
```diff
@@ -59,19 +59,15 @@ const getTaskDetailsByKey = taskKey => {
     throw new Error(`Invalid task ${taskKey} when getting details`)
   }
 
-  return { key: task.key, label: task.label, apiKey: task.name }
+  return { key: task.key, label: task.label, apiKey: task.apiKey, stateKey: task.stateKey }
 }
```
Comment (whitewaterdesign) [inline app/routes/cdo/manage/tasks/generic-task-helper.js:63]: Good change.

**Review by whitewaterdesign — APPROVED** [review]
Great stuff, think this is better than the changes I made before.  👍

**Pull request merged**